firedantic/_async/indexes.py

```diff
@@ -0,0 +1,140 @@
+from logging import getLogger
+from typing import Iterable, List, Optional, Set, Type
+
+from google.api_core.operation_async import AsyncOperation
+from google.cloud.firestore_admin_v1 import (
+    CreateIndexRequest,
+    Index,
+    ListIndexesRequest,
+)
+from google.cloud.firestore_admin_v1.services.firestore_admin import (
+    FirestoreAdminAsyncClient,
+)
+
+from firedantic._async.model import AsyncBareModel, IndexDef, IndexField
+from firedantic._async.ttl_policy import set_up_ttl_policies
+
+logger = getLogger("firedantic")
+
+
+async def list_existing_indexes(
+    client: FirestoreAdminAsyncClient, path: str
+) -> Set[IndexDef]:
+    """
+    List existing database indexes and return a set of them
+    for easy comparison with other indexes
+
+    :param client: The Firestore admin client.
+    :param path: Index path in Firestore.
+    :return: Set of IndexDef tuples
+    """
+    raw_indexes = []
+    request = ListIndexesRequest({"parent": path})
+    operation = await client.list_indexes(request=request)
+    async for page in operation.pages:
+        raw_indexes.extend(list(page.indexes))
+
+    indexes = set()
+    for raw_index in raw_indexes:
+        query_scope = raw_index.query_scope.name
+        fields = tuple(
+            IndexField(name=f.field_path, order=f.order.name)
+            for f in raw_index.fields
+            if f.field_path != "__name__"
+        )
+        indexes.add(IndexDef(query_scope=query_scope, fields=fields))
+    return indexes
+
+
+async def create_compose_index(
```

Comment:
I think I'd prefer this to be called `create_composite_index`.

Reply:
omg, this is a typo I didn't notice can my brain reads it as composite 🤦‍♂️ thanks for noticing

Reply:
literally I didn't even get what are you talking about from the first glance

Reply:
That happens sometimes. That's why we have reviews.